### PR TITLE
[Taxation] Documentation update of Taxation component

### PR DESCRIPTION
**Pull request by tuka217**
| Q             | A
| ------------- | ---
| Doc fix?      | yes
| New docs?  | no
| Fixed tickets   | -

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,8 @@
 build   
 _exts/sensio/__init__.pyc
 *.pyc
+
+/src/
+composer.json
+composer.lock
+/vendor/
```
Comment (TheMadeleine) [inline .gitignore]: ``.gitignore ``should be in global exclude, I think. And I don't get why do you want to ignore these things in a project that does not have them. :)

File: components/Taxation/basic_usage.rst [hunk 2]
```diff
@@ -0,0 +1,142 @@
+Basic Usage
+===========
+
+Tax Rate
+--------
+
+This example show how to create an instance of TaxRate class and introduce usage of the most important properties of
```
Comment (TheMadeleine) [inline components/Taxation/basic_usage.rst]: show**s** [...] introduce**s** [...]

File: components/Taxation/basic_usage.rst [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    $taxCategory->getUpdatedAt();  // return object of class DateTime with string "2015-10-29 11:03:02.000000"
+
+    $taxRate1 = new TaxRate();
+    $taxRate1->setName('taxRate1');
+    $taxRate2 = new TaxRate();
+    $taxRate2->setName('taxRate2');
+
+    $taxCategory->addRate($taxRate1);
+    $taxCategory->addRate($taxRate2);
+    $taxCategory->getRates(); //return collection of object which implements TaxRateInterface (2 object)
```
Comment (TheMadeleine) [inline components/Taxation/basic_usage.rst]: ``//returns a collection of objects that implement the TaxRateInterface``

File: components/Taxation/basic_usage.rst [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+Tax Rate
+--------
+
+Every tax rate has two identifiers, an ID and name. You can access those by calling ``->getId()`` and ``getName()``
+respectively. The name is mutable, so you can change it by calling ``->setName('EU VAT')`` on the tax rate instance.
+
+Tax Amount
+~~~~~~~~~~
+
+An tax rate has two basic amounts. The first is the *amount* and the second is *amount as percentage* (default it is set to 0).
```
Comment (TheMadeleine) [inline components/Taxation/basic_usage.rst]: **A** tax rate has two basic amounts - the*amount* and the *amount as percentage* (by default equal 0).

File: components/Taxation/basic_usage.rst [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+
+.. code-block:: php
+
+    <?
```
Comment (TheMadeleine) [inline components/Taxation/basic_usage.rst]: `<?php`

File: components/Taxation/basic_usage.rst [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+
+    use Sylius\Component\Taxation\Model\TaxRate;
+    use Sylius\Component\Taxation\Model\TaxCategory;
+
+    $taxRate = new TaxRate();
+    $taxCategory = new TaxCategory();
+
+    $taxRate->setAmount(0.5);
+    $taxRate->getAmount(); //Output will be 0.5
```
Comment (TheMadeleine) [inline components/Taxation/basic_usage.rst]: space after //

File: components/Taxation/basic_usage.rst [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+    $taxRate->getAmountAsPercentage(); //Output will be 50
+
+
+Tax Category
+~~~~~~~~~~~~
+
+Every tax rate can have tax category. You can simply set or unset it by calling ``->setCategory()``.
```
Comment (TheMadeleine) [inline components/Taxation/basic_usage.rst]: [...] can have **a** tax category.

File: components/Taxation/basic_usage.rst [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+    $taxRate->setCategory($taxCategory);
+    $taxRate->getCategory(); //Output will be  $taxCategory object
+    $taxRate->setCategory();
+    $taxRate->getCategory(); //Output will be null
+
+
+Including tax rate in price
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can also included tax rate in price by calling ``setIncludedInPrice(true)`` (default it is set to false).
```
Comment (TheMadeleine) [inline components/Taxation/basic_usage.rst]: `You can also include a tax rate in price [...] (false by default).`

File: components/Taxation/basic_usage.rst [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+To check if tax rate is included in price call ``isIncludedInPrice()``.
+
+.. hint::
+    You can read how this property influences on price in chapter :ref:`default-calculator`.
```
Comment (TheMadeleine) [inline components/Taxation/basic_usage.rst]: `[...] influences the price [...]`

File: components/Taxation/basic_usage.rst [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+Tax Category
+------------
+
+Every tax category  has two identifiers, an ID and name. You can access those by calling ``->getId()`` and ``getName()``
+respectively. The name is mutable, so you can change it by calling ``->setName('Clothing')`` on the tax category instance.
+
+Tax Rate Management
+~~~~~~~~~~~~~~~~~~~
+The collection of tax rates (Implementing the ``Doctrine\Common\Collections\Collection`` interface) can be obtained using
+the ``getRates()``. To add or remove tax rates, you can use the ``addRate()`` and ``removeRate()`` methods.
```
Comment (TheMadeleine) [inline components/Taxation/basic_usage.rst]: the ``getRates()`` *method*.

File: components/Taxation/basic_usage.rst [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+    $taxRate->setCategory($taxCategory);
+    $taxRate->getCategory(); // Output will be  $taxCategory object
+    $taxRate->setCategory();
+    $taxRate->getCategory(); // Output will be null
+
+
+Including tax rate in price
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can also included a tax rate in price by calling ``setIncludedInPrice(true)`` (false by default).
```
Comment (pjedrzejewski) [inline components/Taxation/basic_usage.rst]: You can mark tax rate as included in price by

File: components/Taxation/basic_usage.rst [hunk 12]
```diff
@@ -0,0 +1,179 @@
...
+To check if tax rate is included in price call ``isIncludedInPrice()``.
+
+.. hint::
+    You can read how this property influences on price in chapter :ref:`default-calculator`.
+
+
```
Comment (pjedrzejewski) [inline components/Taxation/basic_usage.rst]: Extra blank line.

File: components/Taxation/basic_usage.rst [hunk 13]
```diff
@@ -0,0 +1,179 @@
...
+Including tax rate in price
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can also included a tax rate in price by calling ``setIncludedInPrice(true)`` (false by default).
+To check if tax rate is included in price call ``isIncludedInPrice()``.
+
+.. hint::
+    You can read how this property influences on price in chapter :ref:`default-calculator`.
```
Comment (pjedrzejewski) [inline components/Taxation/basic_usage.rst]: influences the tax calculation

File: components/Taxation/basic_usage.rst [hunk 14]
```diff
@@ -0,0 +1,179 @@
...
+
+Setting calculator
+~~~~~~~~~~~~~~~~~~
+
+To set type of calculator for your tax rate object call ``setCalculator('nameOfCalculator')``. Notice that nameOfCalculator
+should be the same as name of your calculator object.
+
+.. hint::
+    To understand meaning of this property go to :ref:`delegating-calculator`.
+
```
Comment (pjedrzejewski) [inline components/Taxation/basic_usage.rst]: Extra blank line.

File: components/Taxation/basic_usage.rst [hunk 15]
```diff
@@ -0,0 +1,179 @@
...
+
+Tax Category
+------------
+
+Every tax category  has two identifiers, an ID and name. You can access those by calling ``->getId()`` and ``getName()``
+respectively. The name is mutable, so you can change it by calling ``->setName('Clothing')`` on the tax category instance.
+
+Tax Rate Management
+~~~~~~~~~~~~~~~~~~~
```
Comment (pjedrzejewski) [inline components/Taxation/basic_usage.rst]: Missing blank line.

File: components/Taxation/basic_usage.rst [hunk 16]
```diff
@@ -0,0 +1,179 @@
...
+
+.. code-block:: php
+
+    <?php
+
+    use Sylius\Component\Taxation\Model\TaxRate;
+    use Sylius\Component\Taxation\Calculator\DefaultCalculator;
+
+    $taxRate = new TaxRate();
+    $taxRate->setAmount(2);
```
Comment (pjedrzejewski) [inline components/Taxation/basic_usage.rst]: This tax rate is 200%, which does not sound realistic. Paying 20$ tax for a 10$ purchase, that's a bit crazy. :D :penguin:

File: components/Taxation/basic_usage.rst [hunk 17]
```diff
@@ -0,0 +1,179 @@
...
+    use Sylius\Component\Registry\ServiceRegistry;
+    use Sylius\Component\Taxation\Calculator\DelegatingCalculator;
+
+    $taxRate = new TaxRate();
+    $taxRate->setAmount(2);
+    $base = 3; //set base price to 3
+    $defaultCalculator = new DefaultCalculator();
+
+    $serviceRegistry =
+    new ServiceRegistry('Sylius\Component\Taxation\Calculator\CalculatorInterface');
```
Comment (pjedrzejewski) [inline components/Taxation/basic_usage.rst]: ``CalculatorInterface::class``, we are on PHP 5.5 now. :>

File: components/Taxation/models.rst [hunk 18]
```diff
@@ -1,58 +1,55 @@
...
+| amount          | Amount as float (for example 0,23) | float                 |
++-----------------+------------------------------------+-----------------------+
+| includedInPrice | Is the tax included in price?      | bool                  |
++-----------------+------------------------------------+-----------------------+
+| calculator      | Type of calculator                 | string                |
++-----------------+------------------------------------+-----------------------+
+| createdAt       | Date when the rate was created     | \\DateTime            |
++-----------------+------------------------------------+-----------------------+
+| updatedAt       | Date of the last tax rate update   | \\DateTime            |
++-----------------+------------------------------------+-----------------------+
```
Comment (pjedrzejewski) [inline components/Taxation/models.rst]: We do not need the type, it is already in the api.sylius.org docs. :)

**Comment by pjedrzejewski** [comment]
:+1:

**Comment by pjedrzejewski** [comment]
Thank you Ania!

**Pull request merged**